File: colab/logger.py
import uuid
from datetime import datetime
import requests

SESSION = str(uuid.uuid1())
SEQ = 0
LOGS = []
UID = 'unknown'
KEY='OjwoF3m0l20OFidHsRea3ptuQRfQL10ahbEtLa'
epoch = datetime.now().timestamp()


def send_log(right_now=False):
    global epoch, LOGS
    now = datetime.now()
    FOLDER_NAME = '{0:%Y-%m-%d}'.format(now)
    url = 'https://2twhynojr3.execute-api.ap-northeast-1.amazonaws.com/dev/' + FOLDER_NAME
    now = datetime.now().timestamp()
    delta = (now - epoch)
    epoch = now
    if len(LOGS) > 0 and (right_now or delta > 180):
        data = {
            "session": SESSION,
            "uid": UID,
            "logs": LOGS.copy(),
        }
        LOGS.clear()
        headers = {'x-api-key': f'A{KEY}s'}
        r = requests.post(url, headers=headers, json=data)
        if r.status_code != 200:
            print(data)

def log(**kw):
    global SEQ, LOGS, epoch
    now = datetime.now()
    date = now.isoformat(timespec='seconds')
    logdata = dict(seq=SEQ, date=date, **kw)
    LOGS.append(logdata)
    SEQ += 1
    send_log()
    return logdata

def record_login(uid, **kw):
    global UID
    UID = f'{uid}'
    logdata = log(uid=UID, **kw)
    send_log(right_now=True)
        

if __name__ == '__main__':
    record_login(uid='11111', test='test')

File: logger/logger.js

const URL = 'https://2twhynojr3.execute-api.ap-northeast-1.amazonaws.com/dev/';
const KEY = 'OjwoF3m0l20OFidHsRea3ptuQRfQL10ahbEtLa';

const headers = {
    'x-api-key': `A${KEY}s`,
    'Content-Type': 'application/json'
};

var SESSION = String(uuidv4());
var SEQ = 0;
var UID = 'unknown';
var LOGS = [];
var epoch = Date.now();

function send_log(right_now=false){
    var now = Date.now();
    var delta = (now - epoch);
    epoch = now;
    if (LOGS.length > 0 && (right_now || delta > 180)) {
        data = JSON.stringify({
            'session': SESSION,
            'uid': UID,
            'logs': Array.from(LOGS),
        });
        console.log(data)
        
        fetch(URL, {
            method: 'POST', // or 'PUT'
            headers: headers,
            body: data,
        })
        .then(response => response.json())
        .then(data => {
            console.log('Success:', data);
        })
        .catch((error) => {
            console.error('Error:', error);
        });
        LOGS = [];
    }
}

function log(){
    var now = new Date();
    date = now.toISOString();
    logdata = {
        'seq': SEQ, 
        'date': date,
        'uid': result.uid,
        'type': 'typing',
        'code': result.code,
        'keys': result.keys,
        'counts': result.counts,
        'browser': result.browser
    };
    LOGS.push(logdata);
    console.log(LOGS);
    SEQ += 1;
    // send_log();
    return logdata
}

function record_login(){
    UID = result.uid;
    logdata = log()
    send_log(right_now=true)       
}

// if (require.main === module) {
//     record_login({uid:'11111'}, {test:'test'});
// }
